pkg/openfeature/client.go

```diff
@@ -9,6 +9,8 @@ import (
 type IClient interface {
 	Metadata() ClientMetadata
 	AddHooks(hooks ...Hook)
+	SetEvaluationContext(evalCtx EvaluationContext)
+	EvaluationContext() EvaluationContext
```

Comment:
Some other SDKs don't allow the setting of evaluation context directly on the client, but instead take it as a param to the `getClient` method on the top-level API.

Up to you which you think is better for the go SDK.

Reply:
I think this allows for more configurability and it keeps the client constructor terse. Thoughts @james-milligan ?

Reply:
I do nto have any strong opinions here, however im leaning towards @skyerus's changes, it keeps things declarative and prevents a user needing to pass in an empty `EvaluationContext` into `getClient` when one is not needed.